src/Form/FormAutosuggest.jsx

```diff
@@ -0,0 +1,311 @@
+import React, {
+  useEffect, useState,
+} from 'react';
+import PropTypes from 'prop-types';
+import { KeyboardArrowUp, KeyboardArrowDown } from '../../icons';
+import Icon from '../Icon';
+import { Form, IconButton, Spinner } from '../index';
+import useArrowKeyNavigation from '../hooks/useArrowKeyNavigation';
+
+function FormAutosuggest({
+  children,
+  arrowKeyNavigationSelector,
+  ignoredArrowKeysNames,
+  screenReaderText,
+  value,
+  isLoading,
+  errorMessageText,
+  onChange,
+  onSelected,
+  helpMessage,
+  ...props
+}) {
+  const parentRef = useArrowKeyNavigation({
+    selectors: arrowKeyNavigationSelector,
+    ignoredKeys: ignoredArrowKeysNames,
+  });
+  const [isMenuClosed, setIsMenuClosed] = useState(true);
+  const [state, setState] = useState({
+    displayValue: '',
+    errorMessage: '',
+    dropDownItems: [],
+  });
+
+  const setValue = (itemValue, optValue) => {
+    if (value === itemValue) { return; }
+
+    if (onSelected) { onSelected(itemValue); }
+
+    if (optValue !== state.displayValue) {
+      setState(prevState => ({
+        ...prevState,
+        displayValue: optValue,
+      }));
+    }
+  };
+
+  const handleItemClick = (e, optValue) => {
+    setValue(e.target.value, optValue);
+
+    setState(prevState => ({
+      ...prevState,
+      dropDownItems: '',
+    }));
+
+    setIsMenuClosed(true);
+  };
+
+  function getItems(strToFind = '') {
+    let childrenOpt = React.Children.map(children, (child) => {
+      // eslint-disable-next-line no-shadow
+      const { children, ...rest } = child.props;
+
+      const modifiedOpt = React.cloneElement(child, {
+        ...rest,
+        children,
+        value: children,
+        onClick: (e) => handleItemClick(e, children),
+      });
+
+      return modifiedOpt;
+    });
+
+    if (strToFind.length > 0) {
+      childrenOpt = childrenOpt
+        .filter((opt) => (opt.props.children.toLowerCase().includes(strToFind.toLowerCase())));
+    }
+
+    return childrenOpt;
+  }
+
+  const handleExpand = (e) => {
+    setIsMenuClosed(!isMenuClosed);
+
+    const newState = {
+      dropDownItems: [],
+    };
+
+    if (isMenuClosed) {
+      newState.dropDownItems = getItems(e.target.value);
+      newState.errorMessage = '';
+    }
+
+    setState(prevState => ({
+      ...prevState,
+      ...newState,
+    }));
+  };
+
+  const iconToggle = (
+    <IconButton
+      className="pgn__form-autosuggest__icon-button"
+      src={isMenuClosed ? KeyboardArrowDown : KeyboardArrowUp}
+      iconAs={Icon}
+      size="sm"
+      variant="secondary"
+      alt={isMenuClosed ? 'Open the options menu' : 'Close the options menu'}
```

Comment:
Now that Paragon supports i18n, we should ensure all default English strings are marked for translation with `react-intl`. I will file a separate issue for this 😄

Reply:
Ok, thanks